jetson/src/communication/comm.cpp

```diff
@@ -22,31 +23,37 @@ int open_i2c() {
 #define WRITE 1
 
 void read_i2c(int file, uint8_t mcu_addr, uint8_t * data, uint8_t len) {
-    if (ioctl(file, I2C_SLAVE, mcu_addr) < 0) {
-        log(std::string("ERROR: failed to read from i2c bus address: ") + std::to_string(mcu_addr));
-    } else {
-        i2c_smbus_read_i2c_block_data(file, READ, len, data);
-        log(std::string("INFO: read from i2c bus address: ") + std::to_string(mcu_addr));
-        log(std::string("INFO: data: "));
+    int ioctl_val = ioctl(file, I2C_SLAVE, 0x42);
+    log(std::string("INFO: ioctl_val: "));
+    log(std::to_string(ioctl_val));
+    if ( ioctl_val < 0) {
+	    log(std::string("ERROR: failed to read from i2c bus address: "));
+        return;
+    }
 
-        for (int i = 0; i < len; i++){
-            log(std::to_string(data[i]));
-        }
+    i2c_smbus_read_i2c_block_data(file, READ, len, data);
+    log(std::string("INFO: read from i2c bus address: ") + std::to_string(mcu_addr));
+    log(std::string("INFO: data: "));
+
+    for (int i = 0; i < len; i++){
+        char temp[4];
+        snprintf(temp, sizeof temp, "%x", data[i]);
+        log(std::string(temp));
     }
 }
 
-void write_i2c(int file, uint8_t mcu_addr, uint16_t data){
+void write_i2c(int file, uint8_t mcu_addr, uint8_t * data, uint8_t len){
     if (ioctl(file, I2C_SLAVE, mcu_addr) < 0) {
         log(std::string("ERROR: failed to write to i2c bus address:" + mcu_addr));
     }
-    int ret = i2c_smbus_write_word_data(file, WRITE, data);
+    int ret = i2c_smbus_write_i2c_block_data(file, WRITE, len, data);
     if (ret < 0) {
         log(  std::string("ERROR: failed to write to i2c bus address:" + mcu_addr) 
             + std::string("cmd: " + WRITE) 
-            + std::string("data: " + data));
+            + std::string("data: " + *data));
```

Comment:
This isnt going to work -- here you're just printing the first element of the array

you need to iterate over data for loop length of len